File: src/CGO.Web/Scripts/ViewModels/ConcertViewModel.js
var CGO = CGO || {};

CGO.makeConcertViewModel = function(concert) {
    var self = {
        id: ko.observable(concert.id),
        href: ko.observable(concert.href),
        title: ko.observable(concert.title),
        location: ko.observable(concert.location),
        dateAndStartTime: concert.dateAndStartTime,
        isPublished: ko.observable(concert.isPublished)
    };

    return self;
};

CGO.makeConcertModel = function(concertViewModel) {
    return {
        id: concertViewModel.id(),
        href: concertViewModel.href(),
        title: concertViewModel.title(),
        location: concertViewModel.location(),
        dateAndStartTime: concertViewModel.dateAndStartTime,
        isPublished: concertViewModel.isPublished
    };
}

File: src/CGO.Web/ViewModels/Api/ConcertViewModel.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace CGO.Web.ViewModels.Api
{
    public class ConcertViewModel
    {
        public string Href { get; set; }

        public int Id { get; set; }

        [Required]
        public string Title { get; set; }

        // Automatically required, because this is a value type
        public DateTime DateAndStartTime { get; set; }

        [Required]
        public string Location { get; set; }

        public bool IsPublished { get; set; }
    }
}